packages/core/src/tracing/shimmer.js

```diff
@@ -0,0 +1,60 @@
+/*
+ * (c) Copyright IBM Corp. 2023
+ */
+// @ts-nocheck
+
+'use strict';
+
+const shimmer = require('shimmer');
+let logger;
+logger = require('../logger').getLogger('tracing/shimmer', newLogger => {
+  logger = newLogger;
+});
+
+exports.unwrap = shimmer.unwrap;
+
+/**
+ * This wrapper prevents that the application of the customer dies.
+ *
+ * We won't catch errors which happened e.g. when the result of the original fn comes back.
+ * This is fine, because we want to primarly catch errors which happened before the library call.
+ */
```

Comment:
_comment:_ While this wrapper is really super useful, it is also kind of hard to wrap your head around the order in which things are called. WDYT about adding a comment explaining how the flow of control looks?

Something like this maybe?

----

Assuming the following flow for an uninstrumented application:

_application code_ -> _original library function_

After applying instrumentation with the help of this wrapper, the flow will looks like this:

_application code_ -> _(anonymous function in line 24 which we might want to give a name, see below)_ -> `instrumentationWrapperMethod`  (e.g. `shimQuery` from our `mysql` instrumentation) -> `instanaOriginalFunctionWrapper` -> _original library function_.

This serves two purposes:
1. We can insert one central try-catch into the flow that is applied for all our instrumentations.
2. The additional named function `instanaOriginalFunctionWrapper` enables us to inspect the stack trace of errors that are thrown and thus determine whether they have been thrown in our instrumentation code or in the instrumented library.

----

I think this would make it easier to understand this wrapper.

Reply:
Yes sure. I will add this